### PR TITLE
Deploy to NPM

### PR DESCRIPTION
**Pull request by jessedijkstra**
Deploy to NPM through travis and compare the package.json version to the current tag

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,14 @@ cache:
...
+  - "export NPM_PACKAGE_VERSION=$(semver $(cat package.json | jsawk 'return this.version')"
+  - "export GIT_PACKAGE_VERSION=$(semver $(git describe --tags))"
 deploy:
-  provider: script
-  skip_cleanup: true
-  script: script/deploy
+  provider: npm
+  email: service@blendle.com
+  api_key:
+    secure: ZucPRGeP1YJRsSClXe399hdtI01MPfLgiHbopWlR2trnxl1Z5Z6m86Y8TkA2eB+xndLeRNFl33jvnpOxgwwZ3A4RQjUxo2jlpLQTjd69G0OVN9UX5z2F/iBes6D5MDOySQ69UeBnFWWpFqsjPwpavOAl6FJ72WOigfhlCQvRxUBG0TzGp1Bg1+f+glirV+oDeKao20NS/mGhQsTVlnSJP4e0wUdkPd0jFLN0i/kjp013Iw748S1TXZ6UWNtSwOCFRZQKvbzXnelGoDeQ4KNycSIy4T4l5YoQKH7YpMezd8VF4VFxU/H7eH5g7GuJwnV+5y/xqgUbf0bRRKunoQAmyzijmC7961w+rdwtYNSNRcmT0K0eFnRRkOiEhphAizayNdlcD95GTth49cNtc6Rin91YznEa0TRmyj7xRfVrLFXf762iEOp0tngxZfOyA4H8InRXkOByptyUX047DWU2hYYhGJPNoTDE4b5GSlJlJV1tpicHmEE+HFoKqjD0QQmuR0Gd/yPnWczMdImvldBIC2y2Oan3MVhJL555oZhIIIbPU1nylxHIeTc4wzXlwSwQDgAL0+kPPJrLGKuME5FtDVQYFU0mmf98pl7B3EZOXMBA/9t76b6cGueY5VhVpa6NiSJ9s6evyXm83xxzQeJNIOUFyJ5gtR7kYq1Qhmd92S0=
```
Comment (danielsneijers) [inline .travis.yml:14]: Should this be in git? Or is this key supposed to be public?

Comment (jessedijkstra) [inline .travis.yml:14]: This is an encrypted secret:
https://docs.travis-ci.com/user/deployment/npm

Comment (danielsneijers) [inline .travis.yml:14]: Great 👍

**Comment by danielsneijers** [comment]
👍

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+script/
+test/
```
Comment (Klaasvaak) [inline .npmignore]: why ignore script and test?

Comment (jessedijkstra) [inline .npmignore]: Was thinking there's no need for them if you use the module, right?

File: .travis.yml [hunk 3]
```diff
@@ -4,9 +4,14 @@ cache:
     - node_modules
 script:
   - script/test
+before_deploy:
+  - "export NPM_PACKAGE_VERSION=$(semver $(cat package.json | jsawk 'return this.version')"
+  - "export GIT_PACKAGE_VERSION=$(semver $(git describe --tags))"
```
Comment (koenbollen) [inline .travis.yml]: Does this git command work for you? I'd expected something like `git rev-parse --short HEAD`.

**Comment by koenbollen** [comment]
:+1:, as far as I know

File: .npmignore (hunk 2 above)
Comment (Klaasvaak) [inline .npmignore]: It's nice to step into your node dependencies and run the tests

Comment (jessedijkstra) [inline .npmignore]: Will add them again :)

File: .travis.yml (hunk 3 above)
Comment (jessedijkstra) [inline .travis.yml]: Yes, it works perfectly

**Comment by Klaasvaak** [comment]
👍

**Pull request merged**